src/imitation/algorithms/adversarial.py

```diff
@@ -402,28 +382,39 @@ def _build_disc_feed_dict(
         )
 
         # Calculate generator-policy log probabilities.
-        log_act_prob = self._gen_policy.action_probability(obs, actions=acts, logp=True)
+        with th.no_grad():
+            obs_th = th.as_tensor(obs, device=self.gen_algo.device)
+            acts_th = th.as_tensor(acts, device=self.gen_algo.device)
+            _, log_act_prob_th, _ = self.gen_algo.policy.evaluate_actions(
+                obs_th, acts_th
```

Comment:
Usually Torch code only uses `torch.Tensor`s. Mixing numpy arrays and tensors creates more work for the caller, and also makes it easy to accidentally cut out gradients, end up with arrays on the wrong device, etc.

Reply:
(code like this would be simpler if we replaced all our ndarrays with Torch tensors, although that might also create more work when using some of the built-in SB3 evaluation code, which generally takes/produces ndarrays)